framework/src/userobject/ElementUserObject.C

```diff
@@ -52,3 +52,8 @@ ElementUserObject::ElementUserObject(const InputParameters & parameters) :
     addMooseVariableDependency(coupled_vars[i]);
 }
 
+bool
+ElementUserObject::requestedJacobian(unsigned int var) const
+{
+  return var < _requested_jacobian_flag.size() && _requested_jacobian_flag[var];
```

Comment:
How is this ever large? Even with thousands of variables it is at most a few kB!